docs/node/workflow-external-dependencies.md

```diff
@@ -0,0 +1,57 @@
+# External Dependencies
+
+:::caution
+This is an advanced feature and should be used with care.
+:::
+
+Workflows in Temporal may be replayed from the beginning of their history when resumed.
+In order for Temporal to recreate the exact state Workflow code was in, the code is required to be [fully deterministic](/docs/node/determinism).
+To prevent breaking determinism, in the Node SDK, Workflow code runs in an isolated execution environment limited to functionality provided by the SDK.
+
+External Dependencies is an isolation breaking mechanism that allows injecting functions from the main NodeJS environment into a Workflow isolate.
+They are typically used in order to inject custom instrumentation (e.g. logger) functions into the isolate.
+
+## [Injection configuration](https://nodejs.temporal.io/api/modules/worker#injecteddependencyfunction)
+
+The following configuration options are for controlling how an injected function is executed.
+
+### `callDuringReplay`
+
+A boolean controling whether or not the injected function will be called during Workflow replay - defaults to `false`.
+
+### ApplyMode
+
+The different modes for an injected function to be applied to the isolate are documented in the [API reference](https://nodejs.temporal.io/api/enums/worker.applymode).
+
```

Comment:
i like spelling it out for people who are searching and people who are reading the next seciton

```suggestion
The different modes for an injected function to be applied to the isolate are documented in the [API reference](https://nodejs.temporal.io/api/enums/worker.applymode).

- `ASYNC`
- `ASYNC_IGNORED`
- `SYNC`
- `SYNC_IGNORED`
- `SYNC_PROMISE`

```